doc/source/install.rst

```diff
@@ -4,11 +4,12 @@
 Setting up pygfunction
 **********************
 
-*pygfunction* uses Python 3.7, along with the following packages:
+*pygfunction* uses Python 3.8, along with the following packages:
```

Comment:
@MassimoCimmino thanks, this looks great! One minor point - python 3.8 is past end of life, so perhaps in a separate PR this could be bumped up to a more current version.

Reply:
Yes. I tend to update Python versions and dependencies on major and minor releases. This is one place where I had forgotten to update on the last version.

A little summary of my implementation if you want to discuss any of the changes:

1. The `gFunction` class now accepts `Borefield` objects. That did not require changes in the `gFunction` class (outside of `isinstance` checks) since the new class behaves like a list of boreholes for all operations done in pygfunction.
2. The `Borefield.evaluate_g_function` class method returns an array of g-function values. This was to align with the current `gFunction.evaluate_g_function` method.
3. I moved borefield creation functions (e.g. `boreholes.rectangle_field`) to the `Borefield` class. They needed to be re-implemented to generate arrays of geometrical parameters instead of lists of boreholes. The functions in the `boreholes` module now have a deprecation warning.
4. Examples are update to use the new interface for bore field creation.

Something to consider is if the `boreholes.visualize_borefield` should also be moved to the `Borefield` class. I remember we had a discussion about the dependency for matplotlib. An option is to include the method and import matplotlib at runtime within the method instead of the top of the module.

The PR is much larger than I expected. I will review it and make sure it is compatible with the planned changes for issue MassimoCimmino/pygfunction#210.